### ISSUE TITLE
unsupported operand type(s) for &: 'numpy.flagsobj' and 'int'

### ISSUE THREAD

**Issue by takluyver** [h5py/h5py#1400]
attn @kif - I suspect this was introduced by #1390

I just built the latest version, and accessing data from a dataset starts printing error messages like this:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
TypeError: unsupported operand type(s) for &: 'numpy.flagsobj' and 'int'

Exception ignored in: 'h5py.utils.check_numpy'
TypeError: unsupported operand type(s) for &: 'numpy.flagsobj' and 'int'
```

That's it exactly - there's no traceback.

It still returns a result - it is ignoring the error, as it says. But clearly we'll need to fix this.

**Comment by kif** [comment]
> That's it exactly - there's no traceback.
> 
> It still returns a result - it is ignoring the error, as it says. But clearly we'll need to fix this.

I agree: it is very annoying to have all logging mechanism disabled in
h5py by default. The debugging of the tests is very painful but I am
currently working on it.

**Comment by kif** [comment]
I put a fix in #1406

**Comment by takluyver** [comment]
Thanks - I tested #1406 briefly and the messages are indeed gone.
